docs/topics/data-analysis/data-analysis-notebooks-output-formats.md

```diff
@@ -1,35 +1,230 @@
+
+## HTML
+
+Kotlin Notebook can render HTML directly, showing scripts or even embedding websites:
+
+```none
+HTML("""
+<p>Counter: <span id="ctr">0</span> <button onclick="inc()">Increment</button></p>
+<script>
+function inc() {
+let counter = document.getElementById("ctr")
+counter.innerHTML = parseInt(counter.innerHTML) + 1;
+}
+</script>
+""")
+```
+
+![Using HTML script](direct-html-output.png){width=300}
 
 ## Images
 
-Static images can be displayed in formats such as PNG, JPEG, and SVG. Results of the BufferedImage type are also supported. All these images can be images from files, generated graphs, or any other visual media.
+With Kotlin Notebook, you can display images from files, generated graphs, or any other visual media.
+Static images can be displayed in formats such as `.png`, `jpeg`, and `.svg`.
+
+### Buffered images
+
+By default, you can use `BufferedImage` type to display images:
+
+```kotlin
+import java.awt.Color
+import java.awt.image.BufferedImage
+
+val width = 300
+val height = width
+
+val image = BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB)
+
+val graphics = image.createGraphics()
+graphics.background = Color.BLACK
+graphics.clearRect(0, 0, width, height)
+graphics.setRenderingHint(
+    java.awt.RenderingHints.KEY_ANTIALIASING,
+    java.awt.RenderingHints.VALUE_ANTIALIAS_ON
+)
+graphics.color = Color.WHITE
+graphics.fillRect(width / 10, height * 8 / 10, width * 10 / 20, height / 10)
+graphics.dispose()
+```
+
+![Using default BufferedImage to display images](bufferedimage-output.png){width=400}
+
+### Loaded images
+
+With the help of the `lib-ext` library, you can extend the standard Jupyter functionality and display images loaded
+from the network:
+
+```none
+%use lib-ext(0.11.0-398)
+```
+
+```kotlin
+Image("https://kotlinlang.org/docs/images/kotlin-logo.png", embed = false).withWidth(300)
+```
+
+![Using external image links](external-images-output.png){width=400}
+
+### Embedded images
+
+A disadvantage of images loaded from the network is that the image disappears if the link breaks or if you lose
+the network connection. To work around that, use embedded images, for example:
+
+```kotlin
+val kotlinMascot = Image("https://blog.jetbrains.com/wp-content/uploads/2023/04/DSGN-16174-Blog-post-banner-and-promo-materials-for-post-about-Kotlin-mascot_3.png", embed = true).withWidth(400)
+kotlinMascot
+```
+
+![Using embedded images](embedded-images-output.png){width=400}
+
+## Math formulas and equations
+
+You can render mathematical formulas and equations using the LaTeX format, a typesetting system widely used in academia:
+
+1. Add the `lib-ext` library that extends the functionality of the Jupyter kernel to your notebook:
+
+   ```none
+   %use lib-ext(0.11.0-398)
+   ```
+
+2. In the new cell, run your formula: 
+    
+   ```none
+   LATEX("c^2 = a^2 + b^2 - 2 a b \\cos\\alpha")
+   ```
+
+![Using LaTeX to render mathematical formulas](latex-output.png){width=300}
+
+## Data frames
```

Comment:
I discussed it with the DataFrame team, they say that "dataframe" is the variant of spelling they used too. Me too. I'm not sure which variant is correct, just asking you to pay attention

Reply:
Here we use it as a general noun, as opposed to the product name Kotlin DataFrame (below). As I see, the [Kotlin DataFrame](https://kotlin.github.io/dataframe/gettingstarted.html) documentation uses the same approach